tf_agents/agents/ppo/examples/v2/train_eval_rnd.py

```diff
@@ -0,0 +1,301 @@
+  --root_dir=$HOME/tmp/rndppo/gym/PongDeterministic-v0/ \
+  --logtostderr
+```
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import time
+
+from absl import app
+from absl import flags
+from absl import logging
+
+import gin
+import tensorflow as tf
+
+from tf_agents.agents.ppo import rndppo_agent
+from tf_agents.drivers import dynamic_episode_driver
+from tf_agents.environments import parallel_py_environment
+from tf_agents.environments import tf_py_environment
+from tf_agents.environments import suite_gym
+from tf_agents.environments import suite_atari
+from tf_agents.environments.suite_atari import DEFAULT_ATARI_GYM_WRAPPERS_WITH_STACKING
+from tf_agents.environments.atari_preprocessing import AtariPreprocessing
+from tf_agents.environments.atari_wrappers import FrameStack4, DivideBy255
+from tf_agents.eval import metric_utils
+from tf_agents.metrics import tf_metrics
+from tf_agents.networks import actor_distribution_network
+from tf_agents.networks import actor_distribution_rnn_network
+from tf_agents.networks import encoding_network
+from tf_agents.networks import value_network
+from tf_agents.networks import value_rnn_network
+from tf_agents.replay_buffers import tf_uniform_replay_buffer
+from tf_agents.utils import common
+
+
+flags.DEFINE_string('root_dir', os.getenv('TEST_UNDECLARED_OUTPUTS_DIR'),
+                    'Root directory for writing logs/summaries/checkpoints.')
+flags.DEFINE_string('env_name', 'PongDeterministic-v0', 'Name of an environment')
+flags.DEFINE_integer('replay_buffer_capacity', 129,
+                     'Replay buffer capacity per env.')
+flags.DEFINE_integer('num_parallel_environments', 16,
+                     'Number of environments to run in parallel')
+flags.DEFINE_integer('num_environment_steps', 2000000000,
+                     'Number of environment steps to run before finishing.')
+flags.DEFINE_integer('num_epochs', 4,
+                     'Number of epochs for computing policy updates.')
+flags.DEFINE_integer(
+    'collect_episodes_per_iteration', 16,
+    'The number of episodes to take in the environment before '
+    'each update. This is the total across all parallel '
+    'environments.')
+flags.DEFINE_integer('num_eval_episodes', 30,
+                     'The number of episodes to run eval on.')
+flags.DEFINE_boolean('use_rnns', False,
+                     'If true, use RNN for policy and value function.')
+FLAGS = flags.FLAGS
+
+@gin.configurable
+def train_eval(
+    root_dir,
+    env_name='PongDeterministic-v0',
+    env_load_fn=suite_gym.load,
+    random_seed=0,
+    # TODO(b/127576522): rename to policy_fc_layers.
+    actor_fc_layers=(512, 448),
+    value_fc_layers=(512, 448),
+    use_rnns=False,
+    # Params for collect
+    num_environment_steps=2000000000,
+    collect_episodes_per_iteration=16,
+    num_parallel_environments=16,
+    replay_buffer_capacity=129,  # Per-environment
```

Comment:
Why is this 129?

Reply:
I thought this parameter was the rollout length, so I tried to match the RND paper.